### DISCUSSION TITLE
Bugs in Metrics Calculation

### DISCUSSION THREAD

**Discussion by shun-zheng** [PaddlePaddle/PaddleSpatial#36] [BaiduKDDCup2022] [answered]
It seems that the current evaluation code (commit 2de75e85) still has some bugs.
![image](https://user-images.githubusercontent.com/5029717/167121556-73018798-7902-4dc1-a0cc-45be1e4098ae.png)

The correct code should be

`nan_cond = pd.isna(raw_data).any(axis=1)`

Besides, the previous evaluation code is also updated but with a different logic

![image](https://user-images.githubusercontent.com/5029717/167125735-efad5f04-a5a8-4420-917c-1173cf70448e.png)

So, which one should we follow as the final evaluation standard?

**Comment by ramber1836** [comment]
For the first, yes, you're right. Thanks for pointing that out, this will
be fixed soon.
For the second, the negative 'Patv' values will be filtered out by
raw_data['Patv'] < 0, so the zero 'Patv' values should only be taken into
account.

All The Best,
Xinjiang Lu

**Reply by ramber1836** [reply]
The current version will be adopted. @dolphin-zs

**Reply by shun-zheng** [reply]
BTW, there are two metrics.py files, which one should we use as the evaluation standard?

**Reply by DrownFish19** [reply]
@ramber1836 
Could you provide a detail manual for evaluation? According to the current version, I guess we need to write a method named `forecast` in the `predict.py`, which reads the csv data as input and outputs the predicted result. However, I'm not sure if this way is completely correct.

**Reply by ramber1836** [reply]
OK, I will write a manual for the evaluation which mainly focuses on the proper way to implement your code.

**Comment by ramber1836** [comment, marked as answer]
The current one should be adopted.

All The Best,
Xinjiang Lu

**Comment by ramber1836** [comment]
The condition is dedicated to the rows with missing values (not just
missing target values), thus, I prefer to stick to the current version,
i.e. "nan_cond = pd.isna(raw_data).any(axis=1)".
The performance reported in the newly updated report is about 42.xx.

BTW, the performance you reported is promising. That would be great if your
model can keep doing well on the upcoming test data.

All The Best,
Xinjiang Lu
